staging/src/k8s.io/component-base/logs/options.go

```diff
@@ -50,7 +53,8 @@ func (o *Options) Validate() []error {
 
 // AddFlags add logging-format flag
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
-	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, "Set log format")
+	formats := strings.Join(logRegistry.List(), `", "`)
+	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf(`Set format of logs written by component. Options: "%s". Non default options may not support log related Global flags.`, formats))
```

Comment:
"Non default options may not support log related Global flags"

This is a clue, not help :)

I suggest somehow saying, for each option, what it does or doesn't support, possibly by making folks registering the option register a help string.

Reply:
All global flags come from klog, and most are very klog specific and don't translate to other formats. Only few will make sense to be shared between all formats and maybe one is currently supported (-v for verbosity). 

I think for optional alpha feature it's reasonable to just document that we don't support other flags. I would propose to make it more explicit and just say that "Non default options don't support Global log flags". WDYT?

For Beta I would expect to validate if non supported flags are used. Long term instead of registering help string we should support registering flags per format.

Reply:
> "Non default options don't support Global log flags"

OK, sure, if you think people will be able to tell what a global log flag is from reading the help.

If this is alpha I'd expect at the very least for the help text to warn people against using it, and I might actually expect the name of the flag to include "alpha" or "experimental".

Reply:
Would it help to include a list of the current, unsupported, global flags in the help (and note that it may not be comprehensive)? klog doesn't add flags that often, right?

Reply:
it may even be possible to generate the list with klog's AddFlags method

Reply:
Went with approach proposed by @mtaufen and generated list of flags.